public/preload.js

```diff
@@ -64,6 +64,7 @@ contextBridge.exposeInMainWorld('electron', {
   app: {
     hide: () => ipcRenderer.invoke('app:hide'),
     show: () => ipcRenderer.invoke('app:show'),
-    getSystemLanguage: async () => ipcRenderer.invoke('app:getSystemLanguage')
+    getSystemLanguage: async () => ipcRenderer.invoke('app:getSystemLanguage'),
+    setProxySettings: (proxySettings) => ipcRenderer.invoke('app:setProxySettings', proxySettings)
```

Comment:
since `setProxySettings` returns a Promise, shouldn't this function be async?

Reply:
Mhh, I'm not sure actually. Shouldn't we use the `async` prefix only when using `await` in the function body?
And we don't want to await here I think.

BTW, the "hide" and "show" functions also return a promise, like all the other `invoke` calls.

For consistency sake, we should probably remove all the `async` prefixes in the file.

Reply:
I think you're right 👍